packages/web/src/pages/upload-page/fields/TrackMetadataFields.tsx

```diff
@@ -0,0 +1,135 @@
+import { useState } from 'react'
+
+import { GENRES, getCanonicalName, FeatureFlags } from '@audius/common'
+import { useField } from 'formik'
+
+import DropdownInput from 'components/data-entry/DropdownInput'
+import Input from 'components/data-entry/Input'
+import TagInput from 'components/data-entry/TagInput'
+import TextArea from 'components/data-entry/TextArea'
+import UploadArtwork from 'components/upload/UploadArtwork'
+import { getFeatureEnabled } from 'services/remote-config/featureFlagHelpers'
+import { resizeImage } from 'utils/imageProcessingUtil'
+import { moodMap as untypedMoodMap } from 'utils/moods'
+
+import styles from './TrackMetadataFields.module.css'
+
+const moodMap: Record<string, JSX.Element> = untypedMoodMap
+const MOODS = Object.keys(moodMap).map((k) => ({
+  text: k,
+  el: moodMap[k]
+}))
+
+const messages = {
+  genre: 'Pick a Genre'
+}
+
+type TrackMetadataFieldsProps = {
+  /** Whether or not the preview is playing. */
+  playing: boolean
+  // type: 'track' | 'album' | 'playlist'
+  type: 'track'
+}
+
+const TrackMetadataFields = (props: TrackMetadataFieldsProps) => {
+  const [imageProcessingError, setImageProcessingError] = useState(false)
+  const [artworkField, , artworkHelpers] = useField('artwork')
+  const [titleField, titleMeta, titleHelpers] = useField('title')
+  const [genreField, genreMeta, genreHelpers] = useField('genre')
+  const [, moodMeta, moodHelpers] = useField('mood')
+  const [, tagsMeta, tagsHelpers] = useField('tags')
+  const [, descriptionMeta, descriptionHelpers] = useField('description')
+
+  const onDropArtwork = async (selectedFiles: File[], source: string) => {
+    try {
+      let file = selectedFiles[0]
+      file = await resizeImage(file)
+      const storageV2SignupEnabled = getFeatureEnabled(
+        FeatureFlags.STORAGE_V2_SIGNUP
+      )
+      const storageV2UploadEnabled = getFeatureEnabled(
+        FeatureFlags.STORAGE_V2_TRACK_UPLOAD
+      )
+      if (storageV2SignupEnabled || storageV2UploadEnabled) {
+        // @ts-ignore
+        file.name = selectedFiles[0].name
+      }
+      const url = URL.createObjectURL(file)
+      artworkHelpers.setValue({ url, file, source })
+      setImageProcessingError(false)
+    } catch (err) {
+      setImageProcessingError(true)
```

Comment:
prob want to log the error here (unless you were planning to add all the logging at the end of the project which is what i do lol)